src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -191,6 +181,44 @@ public void dispatchRequest(final RestRequest request, final RestChannel channel
         }
     }
 
+    /**
+     * Checks the request parameters against enabled settings for JSONP and error trace support
+     * @param request
+     * @param channel
+     * @return true if the request does not have any parameters that conflict with system settings
+     */
+    boolean checkRequestParameters(final RestRequest request, final RestChannel channel) {
+        // If JSONP is disabled and someone sends a callback parameter we should bail out before querying
+        if (!settings.getAsBoolean(HTTP_JSON_ENABLE, false) && request.hasParam("callback")) {
+            try {
+                XContentBuilder builder = channel.newBuilder();
+                builder.startObject().field("error","JSONP is disabled.").endObject().string();
+                RestResponse response = new BytesRestResponse(FORBIDDEN, builder);
+                response.addHeader("Content-Type", "application/javascript");
+                channel.sendResponse(response);
+            } catch (IOException e) {
+                logger.warn("Failed to send response", e);
+            }
+            return false;
+        }
+
+        // error_trace cannot be used when we disallow emitting stack traces
+        if (!channel.detailedErrorsEnabled() && request.paramAsBoolean("error_trace", false)) {
```

Comment:
I realize the example already here used this form, but I think it is clearer to use `== false` for inverted conditions.